src/java/org/apache/cassandra/utils/TimeUUID.java

```diff
@@ -38,6 +38,11 @@ public TimeUUID(long uuidTimestamp, long lsb)
         this.lsb = lsb;
     }
 
+    public static TimeUUID fromString(String uuidString)
+    {
+        return fromUuid(UUID.fromString(uuidString));
+    }
+
```

Comment:
This method isn't strictly needed since the tests have been adapted to the absence of CASSANDRA-17176 in `ds-trunk`, but I guess we could leave it in place for when that ticket is included?